### PR TITLE
chore: use SDK http client instead of a separate http-client dependency

### LINKED ISSUE #14538: Explore removal of Apache's httpclient dependency from flow-server
### Describe your motivation

Currently vaadin-flow (flow-server) has a dependency to the httpclient (4.5.13) from Apache. The only class that needs it in flow-server [DefaultFileDownloader](https://github.com/vaadin/flow/blob/bf2b84fed9071d399248b3fedfc0c8922c2282c3/flow-server/src/main/java/com/vaadin/flow/server/frontend/installer/DefaultFileDownloader.java) which is only used on build / development time.

### Describe the solution you'd like

Remove the dependency to Apache's httpclient by e.g. using Java 9's HttpClient or by moving of the whole class and its callers to the dedicated maven / gradle base plugin to reduce the dependency on runtime / production mode.

### PR DESCRIPTION
**Pull request by mstahv**
Less dependencies that might cause conflicts in user projects

Fixes #14538

## Type of change

- [ ] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/stats/StatisticsSender.java [hunk 1]
```diff
@@ -281,20 +278,20 @@ String sendStatistics(ObjectNode json) {
...
-            HttpResponse response = client.execute(post);
-            String responseStatus = response.getStatusLine().getStatusCode()
-                    + ": " + response.getStatusLine().getReasonPhrase();
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(postUrl))
+                    .POST(HttpRequest.BodyPublishers.ofString(data))
+                    .header("Content-Type", "application/json").build();
+
+            HttpResponse<String> r = client.send(request,
```
Comment (mcollovati) [inline vaadin-dev-server/src/main/java/com/vaadin/base/devserver/stats/StatisticsSender.java:287]: nit: can we rename the variable to 'response' or something similar?

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/stats/StatisticsSender.java [hunk 2]
```diff
@@ -304,11 +301,12 @@ private static ObjectNode postData(String postUrl, String data) {
...
             // Update the status and return the results
-            result.put(StatisticsConstants.FIELD_LAST_STATUS, responseStatus);
+            result.put(StatisticsConstants.FIELD_LAST_STATUS,
+                    statusCode + ": "); // TODO is reason phrase needed here
             return result;
 
-        } catch (IOException e) {
-            getLogger().debug("Failed to send statistics.", e);
+        } catch (IOException | InterruptedException ex) {
+            getLogger().debug("Failed to send statistics.", ex);
```
Comment (mcollovati) [inline vaadin-dev-server/src/main/java/com/vaadin/base/devserver/stats/StatisticsSender.java:309]: Interrupt flag should probably be reset when catching `InterruptedException`

`Thread.currentThread().interrupt();`

**Review by mcollovati — CHANGES_REQUESTED** [review]
Some minor changes.
The hard-coded timeout settings are a bit opinionated.
It would be good to make them configurable.
Maybe trying to get the value from system properties, otherwise fallback to default?

**Comment by mstahv** [comment]
@mcollovati timeouts were opinionated indeed, copied from some tutorial 😂 I removed them now, going with defaults would be fine I guess? Or were those configurable in the past 🤔

**Comment by mcollovati** [comment]
I was thinking they were configurable through system properties, as the client builder had `useSystemProperties()` setting.
But it seems like there are no properties for the timeouts in httpclient.
So removing them from the PR could be fine.

**Comment by mstahv** [comment]
👌 That was done in https://github.com/vaadin/flow/pull/16897/commits/9becd05f08c1509ea97d11f2823c807e74ef4488

**Comment by mcollovati** [comment]
And this could be interesting to set up tests with proxy: https://github.com/pan-dora/jdk11-httpclient-tests/blob/master/httpclient/src/test/java/net/httpclient/ProxyAuthTest.java

**Comment by knoobie** [comment]
> Another suggested workaround is to set the Proxy-Authorization header in the request.

keep in mind this old security fix :) https://github.com/apache/httpcomponents-client/commit/a572756592c969affd0ce87885724e74839176fb

**Comment by mstahv** [comment]
Would be great if we'd know if and how developers use proxies 😬 Building extensive integration tests for this is probably not worth the investement but some smaller smoke test would be great.

**Comment by mcollovati** [comment]
I propose not adding custom code for handling proxy authentication, but instead updating the documentation at https://vaadin.com/docs/latest/configuration/development-mode/node-js#proxy-settings-for-downloading-the-front-end-toolchain to suggest setting the `-Djdk.http.auth.tunneling.disabledSchemes` if needed

**Review by mcollovati — APPROVED** [review]

**Comment by mcollovati** [comment]
Created vaadin/docs#2475 and #16966

**Review by mcollovati — APPROVED** [review]

**Comment by platosha** [comment]
Would it be feasible to backport this to 24.1? Hilla native compilation is broken by httpclient dependency that brings `commons-logging`, this would help.

**Pull request merged**